miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/MiniAppPermission.kt

```diff
@@ -0,0 +1,41 @@
+package com.rakuten.tech.mobile.miniapp.permission
+
+/** Type of miniapp permission. **/
+enum class MiniAppPermissionType(val type: String) {
+    UNKNOWN("unknown"),
+    LOCATION("location");
+
+    internal companion object {
+
+        internal fun getValue(type: String) = values().find { it.type == type } ?: UNKNOWN
+    }
+}
+
+/** Type of miniapp custom permission. **/
+enum class MiniAppCustomPermissionType(val type: String) {
+    USER_NAME("rakuten.miniapp.user.USER_NAME"),
+    PROFILE_PHOTO("rakuten.miniapp.user.PROFILE_PHOTO"),
+    CONTACT_LIST("rakuten.miniapp.user.CONTACT_LIST");
+
+    internal companion object {
+
+        internal fun getValue(type: String) = values().find { it.type == type }
+    }
+}
+
+internal enum class MiniAppPermissionResult(val type: String) {
+    ALLOWED("Allowed"),
+    DENIED("Denied");
+
+    internal companion object {
+
+        internal fun getValue(isGranted: Boolean) = if (isGranted) ALLOWED else DENIED
+    }
+}
+
+/** Type of miniapp custom permission result. **/
+enum class MiniAppCustomPermissionResult(val type: String) {
+    ALLOWED("Allowed"),
+    DENIED("Denied"),
+    PERMISSION_NOT_AVAILABLE("Permission not available");
+}
```

Comment:
@ts-leojoseph-r @khairul-alam-licon 
Is it the same for both platforms?

Reply:
`PERMISSION_NOT_AVAILABLE` will be used when user will send unknown permission parameter.